networks/1_get_network_communities.py

```diff
@@ -0,0 +1,191 @@
+#
+# This notebook gets network communities for the compendia (PAO1 and PA14) using different thresholds.
+#
+# The output of this notebook are files for each threshold. These files have the following columns:
+# gene id | module id
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %load_ext rpy2.ipython
+import os
+import pandas as pd
+from core_acc_modules import paths
+from rpy2.robjects import pandas2ri
+
+pandas2ri.activate()
+# -
+
+# ## Set user parameters
+#
+# For now we will vary the correlation threshold (`corr_threshold`) but keep the other parameters consistent
+#
+# We will run this notebook for each threshold parameter
+
+# +
+# Params
+corr_threshold = 0.5
+
+# Params for hclust
+# clustering_method is the distance metric used to determine if clusters should be merged
+# https://en.wikipedia.org/wiki/Hierarchical_clustering
+clustering_method = "average"
+
+# Params for cutreeDynamic
+# minimum cluster size
+min_cluster_size = 30
+
+# The higher the value (or if TRUE), the more and smaller clusters will be produced
+deep_split = 2
+
+
+# Output files
+pao1_membership_filename = f"pao1_membership_{corr_threshold}.tsv"
+pa14_membership_filename = f"pa14_membership_{corr_threshold}.tsv"
+# -
+
+# Load expression data
+pao1_compendium_filename = paths.PAO1_COMPENDIUM
+pa14_compendium_filename = paths.PA14_COMPENDIUM
+
+pao1_compendium = pd.read_csv(pao1_compendium_filename, sep="\t", header=0, index_col=0)
+pa14_compendium = pd.read_csv(pa14_compendium_filename, sep="\t", header=0, index_col=0)
+
+print(pao1_compendium.shape)
+pao1_compendium.head()
+
+print(pa14_compendium.shape)
+pa14_compendium.head()
+
+# ## Make adjacency matrix
+
+# Get perason correlation
+# This correlation matrix will represent the concordance
+# between two gene expression profiles
+pao1_corr = pao1_compendium.corr()
+pa14_corr = pa14_compendium.corr()
+
+pao1_corr.head()
+
+# Create adjacency matrix using threshold defined above
+# The adjacency matrix will determine the strength of the connection between two genes
+# If the concordance is strong enough (i.e. above the threshold), then
+# the genes are connected by an edge
+pao1_adj = (pao1_corr.abs() >= corr_threshold).astype(float)
+pa14_adj = (pa14_corr.abs() >= corr_threshold).astype(float)
+
+pao1_adj.head()
+
+# +
+# Save
+# pao1_adj_filename = os.path.join(paths.LOCAL_DATA_DIR, f"pao1_adj_threshold_{corr_threshold}.tsv")
+# pao1_adj.to_csv(pao1_adj_filename, sep="\t")
+# -
+
+# ## Module detection
+# To detect modules, we want to look for genes that are closely related based on the adjacency matrix (i.e. genes that have similar connections)
+#
+# First we need to calculate the topological overlap measure (TOM) using [TOMsimilarity](https://rdrr.io/cran/WGCNA/man/TOMsimilarity.html). The topological overlap of two nodes reflects their similarity in terms of the commonality of the nodes they connect to.
+#
+# * input: adjacency matrix (square symmetric matrix with 0 and 1 entries)
```

Comment:
Yes so if i'm using a threshold of 0.5, two genes with a correlation of 0.9 would be treated the same as 0.51. 

As a non-networks person, my understanding was that to create your network you needed to binarize your correlation matrix. So you start with the correlation between genes that tells you if genes have a similar expression profile. But then to get modules you need to determine if those genes have similar patterns of connections (i.e. if they are connected to the same genes). Then you can cluster based on that similarity information. 

So you need to binarize them to compare the similarity between connections..? Or could you just perform clustering based on the raw correlation score instead?

Reply:
> As a non-networks person, my understanding was that to create your network you needed to binarize your correlation matrix.

You don't necessarily have to, although you can and I think it's fine to start by doing that. The alternative would just be to set everything below your threshold to 0, and keep everything above your threshold as an edge weight (so instead of setting it to 1 you'd keep it as 0.51 or 0.9). Not all clustering methods will work with weighted edges (I'm not sure if the topological overlap you're calculating now would work with edge weights), but there are quite a few that do.